src/main/java/com/uber/cadence/internal/logging/ReplayAwareLogger.java

```diff
@@ -0,0 +1,428 @@
+package com.uber.cadence.internal.logging;
+
+import com.uber.cadence.internal.replay.ReplayAware;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+public class ReplayAwareLogger implements Logger {
+  private final Logger log;
+  private final ReplayAware context;
+  private final Supplier<Boolean> enableLoggingInReplay;
+
+  public ReplayAwareLogger(
+      Logger logger, ReplayAware context, Supplier<Boolean> enableLoggingInReplay) {
+    this.log = logger;
+    this.context = context;
+    this.enableLoggingInReplay = enableLoggingInReplay;
+  }
+
+  @Override
+  public String getName() {
+    return log.getName();
+  }
+
+  @Override
+  public boolean isTraceEnabled() {
+    return log.isTraceEnabled();
```

Comment:
This should include shouldSkipLogging as well as all similar Is...Enabled methods.

Reply:
Done.